src/distances/alignment_helpers.rs

```diff
@@ -0,0 +1,559 @@
+}
+
+// Iterative version of traceback function so we can benchmark both this and recursive option
+pub fn traceback_iterative<T: Number>(
+    table: &Vec<Vec<(usize, Direction)>>,
+    unaligned_seqs: (&[T], &[T]),
+) -> (Vec<T>, Vec<T>) {
+    let mut row_index = table.len() - 1;
+    let mut column_index = table[0].len() - 1;
+
+    let (mut aligned_x, mut aligned_y) = (Vec::<T>::new(), Vec::<T>::new());
+    let (unaligned_x, unaligned_y) = unaligned_seqs;
+    let mut direction = table[row_index][column_index].1;
+
+    while direction != Direction::None {
+        match direction {
+            Direction::Diagonal => {
+                aligned_x.insert(0, unaligned_x[column_index - 1]);
+                aligned_y.insert(0, unaligned_y[row_index - 1]);
+                row_index -= 1;
+                column_index -= 1;
+            }
+            Direction::Left => {
+                aligned_x.insert(0, unaligned_x[column_index - 1]);
+                aligned_y.insert(0, T::from(b'-').unwrap());
+                column_index -= 1;
+            }
+            Direction::Up => {
+                aligned_x.insert(0, T::from(b'-').unwrap());
+                aligned_y.insert(0, unaligned_y[row_index - 1]);
+                row_index -= 1;
+            }
+            Direction::None => {}
+        }
+
+        direction = table[row_index][column_index].1;
+    }
+
+    (aligned_x, aligned_y)
+}
+
+// Public function for recurisve traceback to get alignment and edit distance (ignores ties for now)
+pub fn traceback_recursive<T: Number>(
+    table: &Vec<Vec<(usize, Direction)>>,
+    unaligned_seqs: (&[T], &[T]),
+) -> (Vec<T>, Vec<T>) {
+    let indices = (table.len() - 1, table[0].len() - 1);
+
+    let alignment = _traceback_recursive(table, indices, unaligned_seqs, (Vec::new(), Vec::new()));
+
+    alignment
+}
+
+// Private traceback recursive function. Returns a single alignment (disregards ties for best-scoring alignment)
+fn _traceback_recursive<T: Number>(
+    table: &Vec<Vec<(usize, Direction)>>,
+    indices: (usize, usize),
+    unaligned_seqs: (&[T], &[T]),
+    aligned_seqs: (Vec<T>, Vec<T>),
+) -> (Vec<T>, Vec<T>) {
+    let (unaligned_x, unaligned_y) = unaligned_seqs;
+    let (mut aligned_x, mut aligned_y) = aligned_seqs;
+
+    let (mut row_index, mut column_index) = indices;
+
+    let direction = table[row_index][column_index].1;
+
+    match direction {
+        Direction::Diagonal => {
+            aligned_x.insert(0, unaligned_x[column_index - 1]);
+            aligned_y.insert(0, unaligned_y[row_index - 1]);
+            row_index -= 1;
+            column_index -= 1;
+        }
+        Direction::Left => {
+            aligned_x.insert(0, unaligned_x[column_index - 1]);
+            aligned_y.insert(0, T::from(b'-').unwrap());
+            column_index -= 1;
+        }
+        Direction::Up => {
+            aligned_x.insert(0, T::from(b'-').unwrap());
+            aligned_y.insert(0, unaligned_y[row_index - 1]);
```

Comment:
Same as before about pushing to the end of the vectors and reversing at the end of the outer function